src/global_sequences_viewer.hpp

```diff
@@ -0,0 +1,89 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+#include "model.hpp"
+#include "modelconfig.hpp"
+#include "modelversion.hpp"
+#include "sequence_manager.hpp"
+#include "status.hpp"
+
+namespace ovms {
+
+class GlobalSequencesViewer {
+private:
+    std::mutex mutex;
+    std::map<std::string, SequenceManager*> registeredSequenceManagers;
+
+    /**
+         * @brief sequence Watcher thread for monitor changes in config
+         */
+    void sequenceWatcher(std::future<void> exit);
+
+    /**
+         * @brief A thread object used for monitoring sequence timeouts
+         */
+    std::thread sequenceMonitor;
+
+    /**
+         * @brief An exit signal to notify watcher thread to exit
+         */
+    std::promise<void> exit;
+
+    /**
+         * Time interval between each sequence timeout check
+         */
+    uint sequenceWatcherIntervalSec = 1;
```

Comment:
Maybe we could start with uint32 max value?
Also please use uint32_t

Reply:
I propose: DEFAULT_SEQUENCE_TIMEOUT_SECONDS / 2

Reply:
Nope must be limits.